Hi, Rene, 

I found that when we simply run 'male', make command automatically get into test.
And, in the test program fails because git actions try to run program as root, and
mpirun seems to be required a -allow-run-as-root to run asa root.

One approach is adding an option to define mpirun and options in configure, 
but I want to ask how do you treat mpirun in ASPECT.

Thanks,

Here is the part of the log.

# Test in Dynamobench_case1 directory...
make[3]: Entering directory '/__w/calypso/calypso/tests/Dynamobench_case1'
rm -f *.dat
mpirun -np 4 /__w/calypso/calypso/bin/gen_sph_grids
--------------------------------------------------------------------------
Makefile:27: recipe for target 'test' failed
mpirun has detected an attempt to run as root.
make[3]: Leaving directory '/__w/calypso/calypso/tests/Dynamobench_case1'
# Test in Dynamobench_case2 directory...
Running at root is *strongly* discouraged as any mistake (e.g., in
defining TMPDIR) or bug can result in catastrophic damage to the OS
file system, leaving your system in an unusable state.

You can override this protection by adding the --allow-run-as-root
option to your cmd line. However, we reiterate our strong advice
against doing so - please do so at your own risk.
--------------------------------------------------------------------------

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -6,25 +6,20 @@ concurrency:
   group: ${{ github.actor }}-${{ github.ref }}
   cancel-in-progress: true
 
+env:
+  # Avoid a warning
+  OMPI_MCA_btl_base_warn_component_unused: 0
+  # Prevent slowdown when oversubscribing (e.g. 4 ranks of 2 cores)
+  OMPI_MCA_mpi_yield_when_idle: 1
+  OMPI_MCA_rmaps_base_oversubscribe: 1
+  # Allow run as root
+  OMPI_ALLOW_RUN_AS_ROOT: 1
+  OMPI_ALLOW_RUN_AS_ROOT_CONFIRM: 1
+  # Fix OpenMPI issue in Docker : https://github.com/open-mpi/ompi/issues/4948
+  OMPI_MCA_btl_vader_single_copy_mechanism: none
+
 jobs:
-  container-compile:
-    # linux build
-    name: test-compile
-    runs-on: [ubuntu-20.04]
-    container:
-      image: geodynamics/calypso-buildenv-bionic:latest
-    steps:
-    - uses: actions/checkout@v2
-    - name: build
-      run: |
-        ./configure \
-          --enable-fftw3 \
-          --with-hdf5 \
-          --with-blas
-        # Fix OpenMPI issue in Docker : https://github.com/open-mpi/ompi/issues/4948
-        export OMPI_MCA_btl_vader_single_copy_mechanism=none
-        make
-  build-from-scratch:
+  build-and-test:
     name: test-compile
     runs-on: [ubuntu-20.04]
     steps:
@@ -39,6 +34,4 @@ jobs:
           --with-hdf5 \
           --with-blas
 
-        # Fix OpenMPI issue in Docker : https://github.com/open-mpi/ompi/issues/4948
-        export OMPI_MCA_btl_vader_single_copy_mechanism=none
         make

